collection+xattrs.go

```diff
@@ -618,16 +619,74 @@ func removeUserXattrs(xattrs semiParsedXattrs) {
 	}
 }
 
+// macroExpand takes the macro and assigned 'value' the correct value based on what the macro is provided (this is gocb macro format)
+func (e *event) macroExpand(value string, casServerFormat string, bucketUUID string) string {
+	if value == "\"${Mutation.CAS}\"" {
+		value = casServerFormat
+	} else if value == "\"${Mutation.value_crc32c}\"" {
+		value = encodedCRC32c(e.value)
+	} else {
+		// if the value isn't to be macro expanded to CAS or CRC hash then it is a source ID
+		value = bucketUUID
```

Comment:
It would be safer to have an explicit string for this case too, like `${Mutation.bucketUUID}`, and fail with an error if `value` doesn't match a known case.